app/controllers/adopter_waitlists_controller.rb

```diff
@@ -0,0 +1,70 @@
+class AdopterWaitlistsController < ApplicationController
+  before_action :edit_my_adopters_user, only: %i(create update destroy)
+  before_action :edit_all_adopters_user, only: %i(create update destroy)
+  before_action :load_waitlist, only: %i(create)
+  before_action :load_adopter_waitlist, only: %i(show update destroy)
+  before_action :load_adopter, only: %i(create)
+
+  respond_to :html, :json
+
+  def index
+    redirect_to :root
+  end
+
+  def create
+    @adopter_waitlist = AdopterWaitlist.find_or_initialize_by(create_params)
+
+    if AdopterWaitlist.count == 0
+      @adopter_waitlist.rank = 1
+    else
+      @adopter_waitlist.rank = AdopterWaitlist.maximum('Rank') + 1
```

Comment:
same here, you want the max rank for the `wishlist_id` you're working with, not the whole table.